### PR TITLE
Add an access method to decode the Header x5c field into DER PKIX format.

### PR DESCRIPTION
**Pull request by jnicholls**
Another take on #181 that properly base64-decodes the X.509 certificate chain.

### CODE REVIEW

**Comment by Keats** [comment]
> Ensures the new x5c field, as well as all the other optional RFC7515 fields, are able to be missing when deserializing a Header.

That's already the case...? See all the tests and pretty much every user of the crate

**Comment by jnicholls** [comment]
Yes I was being overly explicit, sorry. The MissingFieldDeserializer handles the Option case already. The reason I thought to add it is when you provide your own `deserialize_with` implementation, the MissingFieldDeserializer is no longer used in the `serde_derive` generated code, so I had to add `#[serde(default)]` explicitly.

I had already worked to remove that stuff as you pointed this out, it's updated now.

File: src/header.rs [hunk 1]
```diff
@@ -1,9 +1,13 @@
-use serde::{Deserialize, Serialize};
+use std::result;
+
+use serde::{de, Deserialize, Serialize};
 
 use crate::algorithms::Algorithm;
 use crate::errors::Result;
 use crate::serialization::b64_decode;
 
+type X509Chain = Vec<Vec<u8>>;
```
Comment (Keats) [inline src/header.rs:9]: Doesn't this type need to be public? I think I would rather skip it tbh as it's only used in this file and the actual type is clearer

**Review by Keats — CHANGES_REQUESTED** [review]

File: src/header.rs (hunk 1 above)
Comment (jnicholls) [reply src/header.rs]: The public-with-private type checker automatically replaces the type with private type aliases in these types of cases when the aliased type is public. I removed it.

**Comment by olitha** [comment]
I wonder if it Is necessary to decode base64 when deserializing. There are other fields already like x5t which is base64 (this time base64url) and is not decoded by serde. On the one hand it is good to hide these details (the exact coding) away from the user, on the other hand there is processing overhead even for fields that are not used by application code.

As an idea. What about classical `impl Header` with getters that decode? This would remove processing *and* hide the encoding details. But would also leave the user being able to get the original values.

**Comment by jnicholls** [comment]
@olitha I tend to agree, albeit there is still deserialization overhead w/ the allocation of the intermediate Strings, just not nearly as much as what base64 decoding introduces, particularly if the application wants to make use of the field as it will always want the DER representation for any practical X.509 use case. The base64 decoder looks to be quite efficient (several GB/s on commodity hardware). Nevertheless, the best argument is to pay the cost only when it's needed, so I will revise.

**Comment by Keats** [comment]
I like that!

**Comment by jnicholls** [comment]
Yes I will add a test. Is in header.rs okay?

**Comment by Keats** [comment]
yep

**Comment by jnicholls** [comment]
On second thought, I might put them into a `tests/header.rs` module. I want to have a legit certificate `.der` file to include into one of the tests, rather than put it into the code directly. I'll move in that direction, but let me know if you have any objections to that.

**Pull request merged**